modules/api/src/main/java/org/apache/ignite/table/mapper/DefaultColumnMapper.java

```diff
@@ -21,7 +21,7 @@
 import org.jetbrains.annotations.NotNull;
 
 /**
- * Simple mapper implementation that map a column to a class field.
+ * Mapper implementation which maps object fields to the columns by their names.
```

Comment:
Which one of the following mappings do we want to achieve:
1. Field -> Column
2. Column -> Field
3. All of the above?

Javadoc says field -> column, but `mappedField` does column -> field, and `mappedColumn` is not clear at all.

Reply:
For now, we allow only one-to-one mapping. (otherwise 'object <-> row' translation looks ambiguous)

So, seems, we will need both methods to resolve: column->field and field->column.
As columns must be written to a row in order, we need dummy-mapping for missed columns (to write 'null' or 'column default'). Resolving 'column->field' allow us to write columns in right order and write a row in one iteration.

On the client side, we may not know all the columns and their defaults.
Resolving 'field->column' allow here to write all fields to a tuple (or whatever client protocol sends to a server in MsgPack format).
However, I've no idea how to deal with orphan fields here (when no column exists that matches field name).
* should we pass them to a server and then bypass validation and ignore orphans.
* disallow orphans and fails with SchemaMismatch exception.

Reply:
I think we should have two-way mapping, column -> field and field -> column.

We know the schema on the client side (column names and their order). We don't know defaults for now, but I don't think they are required.

It should be easy to write the row in correct order from an object if we have `column -> field` mapping:

```
for (var col : schema.columns()) {
    var fieldName = mapper.fieldForColumn(col.name());
    if (fieldName == null)
        packNil();
    else
        packValue(...);
}
```